nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/factory/Nd4jBackend.java

```diff
@@ -156,14 +157,12 @@ public static Nd4jBackend load() throws NoAvailableBackendException {
         String logInitProperty = System.getProperty(ND4JSystemProperties.LOG_INITIALIZATION, "true");
         boolean logInit = Boolean.parseBoolean(logInitProperty);
 
-        List<Nd4jBackend> backends = new ArrayList<>(1);
-        ServiceLoader<Nd4jBackend> loader = ServiceLoader.load(Nd4jBackend.class);
+        List<Nd4jBackend> backends = new ArrayList<>();
+        ServiceLoader<Nd4jBackend> loader = ND4JClassLoading.loadService(Nd4jBackend.class);
```

Comment:
@hosuaby you might want to also change the loadLibrary call in Nd4jBackend. Could you go through and look through all the uses of classloaders and make sure whatever you do is propagated so the classloader takes effect everywhere?

Reply:
Done. PR updated and rebased on master